python/cuml/dask/naive_bayes/naive_bayes.py

```diff
@@ -232,7 +236,48 @@ def predict(self, X, delayed=True, parallelism=5):
         dask.Array containing class predictions
         """
 
-        return self._predict(X, delayed, parallelism)
+        gpu_futures = self.client_.sync(extract_arr_partitions, X)
+        x_worker_parts = workers_to_parts(gpu_futures)
+
+        key = uuid1()
+        futures = [(wf[0],
+                    self.client_.submit(MultinomialNB._get_size,
```

Comment:
Could DistributedDataHandler.calculate_parts_to_sizes handle this?

Reply:
It can, and I originally had refactored this to use that, however I was getting segfaults during benchmarks from the use of sparse arrays. I reverted the changes for now, but have created https://github.com/rapidsai/cuml/issues/1834 so that this refactor can be done in a different PR